gst/nnstreamer/include/nnstreamer_plugin_api_if.h

```diff
@@ -0,0 +1,43 @@
+/* SPDX-License-Identifier: LGPL-2.1-only */
+/**
+ * GStreamer/NNStreamer Tensor-IF
+ * Copyright (C) 2020 Gichan Jang <gichan2.jang@samsung.com>
+ */
+/**
+ * @file	nnstreamer_plugin_api_if.h
+ * @date	28 Oct 2020
+ * @brief	NNStreamer APIs for tensor_if custom condition
+ * @see		https://github.com/nnstreamer/nnstreamer
+ * @author	Gichan Jang <gichan2.jang@samsung.com>
+ * @bug		No known bugs except for NYI items
+ *
+ */
+
+#ifndef __NNS_PLUGIN_API_IF_H__
+#define __NNS_PLUGIN_API_IF_H__
+
+#include <stdint.h>
+#include "tensor_typedef.h"
+
+G_BEGIN_DECLS
+/**
+ * @brief Calls the user defined conditions function.
+ * @return TRUE if there is no error
+ */
+typedef gboolean (* tensor_if_custom) (const GstTensorsInfo *info,
+    const GstTensorMemory *input, gboolean *result);
+
+/**
+ * @brief Register the custom callback function.
+ */
+extern void
+nnstreamer_if_custom_register (GstElement *tensorif_h, tensor_if_custom func, void *data);
+
+/**
+ * @brief Unregister the custom callback function.
```

Comment:
This is internal API. You need to fully supply doxygen info: all parameters and return values.